src/System.IO/tests/Stream/Stream.CopyToTests.cs

```diff
@@ -0,0 +1,338 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class StreamCopyToTests
+    {
+        [Fact]
+        public void IfCanSeekIsFalseLengthAndPositionShouldNotBeCalled()
+        {
+            var baseStream = new DelegateStream(
+                canReadFunc: () => true,
+                canSeekFunc: () => false,
+                readFunc: (array, index, length) => 0); // (buffer, offset, count) declared in enclosing scope
+            var trackingStream = new CallTrackingStream(baseStream);
+
+            var dest = Stream.Null;
+            trackingStream.CopyTo(dest);
+
+            Assert.True(trackingStream.TimesCalled(nameof(trackingStream.CanSeek)) <= 1);
```

Comment:
You could use Assert.InRange here and in other such places, e.g.
```C#
Assert.InRange(trackingStream.TimesCalled(nameof(trackingStream.CanSeek)), 0, 1);
```
The nice thing about that is the message that'll be generated on failure includes both the allowed range and the resulting value that was out of range.